### PR TITLE
Added additional tests for generic attributes.

### LINKED ISSUE #73735: Generic attributes lack testing
We need reflection tests that test generic attributes.

https://github.com/dotnet/linker/issues/2963 made it clear that we don't have any targeted reflection tests for the new scenario in the libraries partition. The newly added test is not a targeted reflection test - it's a S.R.Metadata test.

### PR DESCRIPTION
**Pull request by koenigst**
- Added a test for GetCustomAttribute to verify that the values of the generic type are correctly initialized.
- Added some theory data to check the equality of generic attributes.

Fixes #73735

### CODE REVIEW

File: src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Attributes.cs [hunk 1]
```diff
@@ -298,29 +298,41 @@ public static void GetCustomAttributesWorksWithOpenAndClosedGenericTypesForEvent
...
             Attribute[] attributes = Attribute.GetCustomAttributes(typeof(HasGenericAttribute), typeof(GenericAttribute<>));
-            Assert.Equal(3, attributes.Length);
-            Assert.Equal(1, attributes.Count(a => a.GetType() == typeof(DerivesFromGenericAttribute)));
-            Assert.Equal(1, attributes.Count(a => a.GetType() == typeof(GenericAttribute<bool>)));
-            Assert.Equal(1, attributes.Count(a => a.GetType() == typeof(GenericAttribute<string>)));
+            Assert.Contains(typeof(DerivesFromGenericAttribute), attributes.Select(a => a.GetType()));
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/56887", TestRuntimes.Mono)]
```
Comment (buyaa-n) [inline src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Attributes.cs:308]: Looks https://github.com/dotnet/runtime/issues/56887 is resolved, could you please remove all the `[ActiveIssue("https://github.com/dotnet/runtime/issues/56887", TestRuntimes.Mono)]`?

**Review by buyaa-n — COMMENTED** [review]
Overall looks good thanks @koenigst, lets see if anything fails after removing `[ActiveIssue("https://github.com/dotnet/runtime/issues/56887", TestRuntimes.Mono)]` in the tests

**Comment by koenigst** [comment]
@buyaa-n It looks like `System.Tests.AttributeGetCustomAttributes.GetCustomAttributesOnOpenGenericBaseTypeRetrievesDerivedAttributes` is still [failing on Mono](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-106745-merge-8b970d6d1dad4d39b5/System.Runtime.Tests/1/console.a0cf67b5.log?helixlogtype=result). I will put the active issue attribute back on this test method.

File: src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Attributes.cs [hunk 2]
```diff
@@ -298,29 +291,40 @@ public static void GetCustomAttributesWorksWithOpenAndClosedGenericTypesForEvent
 
         [Fact]
         [ActiveIssue("https://github.com/dotnet/runtime/issues/56887", TestRuntimes.Mono)]
```
Comment (buyaa-n) [inline src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Attributes.cs:293]:
```suggestion
        [ActiveIssue("https://github.com/dotnet/runtime/issues/107095", TestRuntimes.Mono)]
```

**Comment by buyaa-n** [comment]
> @buyaa-n It looks like `System.Tests.AttributeGetCustomAttributes.GetCustomAttributesOnOpenGenericBaseTypeRetrievesDerivedAttributes` is still [failing on Mono](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-106745-merge-8b970d6d1dad4d39b5/System.Runtime.Tests/1/console.a0cf67b5.log?helixlogtype=result). I will put the active issue attribute back on this test method.

Yep, I filed a new issue for that and updated the attribute, thanks!

**Review by buyaa-n — APPROVED** [review]
LGTM, thank you @koenigst!

**Pull request merged**